spec/workflow_service_spec.rb

```diff
@@ -27,56 +27,48 @@
   before(:each) do
     @repo  = 'dor'
     @druid = 'druid:123'
-    @mock_resource = double('mock_rest_client_resource')
-    @mock_http_connection = double('mock_http_connection')
-    @mock_logger = double('Logger')
-
-    allow(@mock_logger).to receive(:info)  # silence log output
-    allow(@mock_logger).to receive(:debug) # silence log output
-    allow(Dor::WorkflowService).to receive(:default_logger).and_return(@mock_logger)
-
+    @mock_resource = RestClient::Resource.new('https://localhost/workflow')
+    @mock_logger   = double('Logger')
     allow(@mock_resource).to receive(:[]).and_return(@mock_resource)
     allow(@mock_resource).to receive(:options).and_return( {} )
-    allow(@mock_resource).to receive(:url).and_return( 'https://localhost/workflow' )
-
+    allow(@mock_logger).to receive(:info)
+    allow(@mock_logger).to receive(:warn)
+    allow(@mock_logger).to receive(:debug) # silence logging output
+    @mock_http_connection = double('mock_http_connection')
     allow(Faraday).to receive(:new).and_return(@mock_http_connection)
-
-    allow(RestClient::Resource).to receive(:new).and_return(@mock_resource)
-    Dor::WorkflowService.configure @mock_resource.url
+    @dwfs = Dor::WorkflowService.new(@mock_resource, :logger => @mock_logger)
   end
 
+  # it 'xxx' do
+  #   require 'pry-byebug'
+  #   binding.pry
+  # end
+
   describe '#create_workflow' do
     it 'should pass workflow xml to the DOR workflow service and return the URL to the workflow' do
       expect(@mock_resource).to receive(:put).with(wf_xml_label, anything).and_return('')
-      Dor::WorkflowService.create_workflow(@repo, @druid, 'etdSubmitWF', wf_xml)
+      @dwfs.create_workflow(@repo, @druid, 'etdSubmitWF', wf_xml)
     end
 
     it 'should log an error and retry upon a targetted RestClient exception, raise on an unexpected Exception' do
-      ex = RestClient::Exception.new(nil, 418)
-      ex.message = "I'm A Teapot"
+      ex = Faraday::ClientError.new RestClient::Exception.new(nil, 418) # I'm a Teapot
       runs = 0
       expect(@mock_resource).to receive(:put).twice {
         runs += 1
         raise ex if runs == 1
         raise Exception.new('Something Else Happened') if runs == 2
       }
-      expect(@mock_logger).to receive(:warn).with(/\[Attempt 1\] RestClient::Exception: #{ex.message}/)
-      expect{ Dor::WorkflowService.create_workflow(@repo, @druid, 'etdSubmitWF', wf_xml) }.to raise_error(Exception, 'Something Else Happened')
+      expect(@mock_logger).to receive(:warn).with(/\[Attempt 1\] Faraday::ClientError/)
+      expect{ @dwfs.create_workflow(@repo, @druid, 'etdSubmitWF', wf_xml) }.to raise_error(Exception, 'Something Else Happened')
     end
```

Comment:
Set the retry parameter on init for Faraday and set the logger on Faraday and leave all of this up to it, no need to spec basic HTTP client behavior here.